### ISSUE TITLE
select:modified state shows unmodified sources have changed even though nothing env-specific is in the source

### ISSUE THREAD

**Issue by nazaniiin89** [fishtown-analytics/dbt#3201]
### Describe the bug
When the `database` parameter in `dev` target in `profiles.yml` is different from state files, the following command returns all source files as modified files even though they are not modified:
`dbt ls --target dev -s state:modified --state .`

I found a couple of bug reports of the same behaviour for when `target` is explicitly included in source definition and it's different between different environments, but I believe this is not the case for me as there is no target included in my source definition (example in Steps to Reproduce). 

[I understand that there could be false positives when there are env-specific parameters in source](https://docs.getdbt.com/reference/node-selection/state-comparison-caveats#false-positives), but is it expected to have false positives even when env-specific parameters are not explicitly included in source syntax?

### Steps To Reproduce
Here is an example of a source file we have:
```
sources:
  - name: etl
    tables:
      - name: etl_sample_table
        identifier: sample_table
models:
  - name: sample_table
```

Here is the command I run to see the list of changes:
`dbt ls --target dev -s state:modified --state .`

These sources appear as modified files when I run the `ls` command. When I run `dbt run --models state:modified --defer --state .`, models from the source files do not run. But when I run `dbt run --models state:modified+ --defer --state .`, then models from source files run.

### Expected behavior
I expect to see no changes in the list of modified files because I did not change any source information.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [x] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.19.0
```

**The operating system you're using:**
MacOS Mojave

**The output of `python --version`:**
`Python 3.6.5`

**Comment by jtcohen6** [comment]
@nazaniiin89 Thanks for opening this issue, and for the detailed writeup! This was bewildering to me initially, but I think I know what's going on.

When using dbt with Snowflake—or BigQuery, or any data platform that offers access to multiple logical databases/projects—all `sources`, unless otherwise specified, use the same `database` as your profile. If your `target.database` is `my_db`, the source's database is `my_db`; if `my_other_db`, then `my_other_db`.

So defining a source without specifying a `database` is effectively the same as defining it as:
```yml
sources:
  - name: etl
    database: "{{ target.database }}"
    tables:
      - name: etl_sample_table
        identifier: sample_table
```

In that sense, the source is using an env-specific parameter, and it will look different in `manifest.json` depending on the `target.database`:

```json
        "source.my_project.etl.etl_sample_table": {
            "fqn": [
                "my_project",
                "etl",
                "etl_sample_table"
            ],
            "database": "dbt_test",
            "schema": "etl",
```
vs.
```
        "source.my_project.etl.etl_sample_table": {
            "fqn": [
                "my_project",
                "etl",
                "etl_sample_table"
            ],
            "database": "dbt_test_alt",
```
And so:
````
$ dbt ls -s state:modified --state .state
source:my_project.etl.etl_sample_table
```

Do you current structure your CI such that the same source tables exist in different logical databases / environments? Are the source tables in your CI database a sample/subset?

In any case, there is an open issue (#2744) to resolve the env-based "false positives" in source definitions. This definitely feels like a case where we want to treat the sources as unmodified, since the env-based parameter is implicit.

**Comment by nazaniiin89** [comment]
@jtcohen6 Thank you for clarifying this through examples. It's now clear to me why I was experiencing this with the state commands.

> Do you current structure your CI such that the same source tables exist in different logical databases / environments? 

Our CI is set up in a way that we get the state files of production database (no sampling) but write results of `dbt run --models state:modified --defer --state .` to test database, which is the `target` database. So, to answer your question, yes the environments we read from and write to are different. From what you explained, I believe for now it's better for the target database to be the same as state database until the issue you linked is resolved. Thanks again.

**Comment by jtcohen6** [comment]
@nazaniiin89 Just to clarify: If the _raw source tables_ that dbt is reading from are always in a certain database—whether that's a database called `raw` or `prod` or anything else—you can add that database to the source definition:
```yml
sources:
  - name: etl
    database: raw
    tables:
      - name: etl_sample_table
        identifier: sample_table
```
If, however, you need to red from _different_ raw source tables in prod vs. CI runs, then a static specification isn't possible.

**Comment by nazaniiin89** [comment]
> @nazaniiin89 Just to clarify: If the _raw source tables_ that dbt is reading from are always in a certain database—whether that's a database called `raw` or `prod` or anything else—you can add that database to the source definition:
> 
> ```yaml
> sources:
>   - name: etl
>     database: raw
>     tables:
>       - name: etl_sample_table
>         identifier: sample_table
> ```
> 
> If, however, you need to red from _different_ raw source tables in prod vs. CI runs, then a static specification isn't possible.

Ah, I see! Yeah, in my case, all sources are in one env and can be read from there. I updated source by explicitly adding the database there and ran the `ls` command and this time I saw no source tables in the result 👍 Thank you, this helps.

**Comment by jtcohen6** [comment]
Nice! Ok, not saying that this issue is resolved, but I just left a comment on #2744 to make sure it handles this case as well. I'm going to close it in favor of that issue in the meantime.
